### ISSUE TITLE
Schema Rules have unhelpful summary in default CLI output

### ISSUE THREAD

**Issue by philsturgeon** [stoplightio/spectral#116]
Linting this document:

```yaml
openapi: 3.0.8
info:
  title: occaecat ex exercitation consequat
  version: sunt Duis aliqua qui
x-testcase: 'should not work without required property: paths'
```

Results in stylish format output:

```
linting ../openapi3-examples/fail/fuzz1/331be1bf-781d-407f-93d6-1f4b390ae32b.yaml
OpenAPI 3.x detected

/home/mike/node/openapi3-examples/fail/fuzz1/331be1bf-781d-407f-93d6-1f4b390ae32b.yaml
 1:1    error  oas3-schema       Validate structure of OpenAPIv3 specification
 1:1  warning  api-servers       OpenAPI `servers` must be present and non-empty array
 2:6  warning  info-contact      Info object should contain `contact` object
 2:6  warning  info-description  OpenAPI object info `description` must be present and non-empty string

✖ 4 problems (1 error, 3 warnings, 0 infos)
```

But the json format output gives the more helpful message for the error:

```json
	{
		"code": "oas3-schema",
		"path": [],
		"message": "should have required property 'paths'",
		"severity": 0
	},
```

_Originally posted by @MikeRalphson in https://github.com/stoplightio/spectral/pull/112#issuecomment-477554960_

**Comment by philsturgeon** [comment]
Yeah this is a tough one. Most of the time the summary is a bit more concise but for oasX-schema based rules you really want the message. Id rather not make the formatters be aware of such things, so we need to find somewhere slightly higher up to make that change perhaps.

**Comment by MikeRalphson** [comment]
> Id rather not make the formatters be aware of such things

Understood, just to clarify, my concern was that a program consuming the json format output cannot obtain the same error message as is displayed by the stylish formatter currently.

**Comment by philsturgeon** [comment]
Oh thats the inverse of what I said! Ha. We can sort that at the same time.

**Comment by philsturgeon** [comment]
@MikeRalphson great so @P0lip just added summary to JSON in #117 so we'll sort the bit I said out later.

**Comment by P0lip** [comment]
Fixed in 2.4.0
